torchao/quantization/quant_primitives.py

```diff
@@ -267,6 +267,11 @@ def _quantize_affine_no_dtype_cast(
     if zero_point is not None:
         zero_point = zero_point.view(shape_after_reduction)
 
+    if zero_point_domain is None:
```

Comment:
would it be better to check `zero_point` here? it seems that we want to say if there is no zero point, maybe zero_point_domain is None can be an assert inside the branch